commit dd574f7ca6097e0032ab1336e6191eea67fb9bac
Author: John Levon <john.levon@joyent.com>
Date:   2019-07-08T21:17:16+00:00 (3 months ago)
    
    MANTA-2843 Manta zones lacking CPU cap thwart provisioning
    TRITON-1801 update deps/eng for sdc-cnapi

diff --git a/config/test.json b/config/test.json
index 62b1df1..eb2398b 100644
--- a/config/test.json
+++ b/config/test.json
@@ -36,6 +36,7 @@
   "dapi": {
       "changeDefaults": {
           "server_spread": "",
+          "filter_capness": "true",
           "filter_headnode": "false",
           "filter_min_resources": "false",
           "filter_large_servers": "",
diff --git a/deps/eng b/deps/eng
index 4e33bf4..940d8fb 160000
--- a/deps/eng
+++ b/deps/eng
@@ -1 +1 @@
-Subproject commit 4e33bf4e26bc6cc8d93b3851fed97c11f513bfe4
+Subproject commit 940d8fbbef6f4d0e6a9e45409fc4646eda8255e8
diff --git a/docs/index.md b/docs/index.md
index 7f65aa4..aaec2ba 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -10,7 +10,7 @@ markdown2extras: tables, code-friendly
 -->
 
 <!--
-    Copyright (c) 2019, Joyent, Inc.
+    Copyright 2019 Joyent, Inc.
 -->
 
 <!--
@@ -73,6 +73,7 @@ found in sapi_manifests/cnapi/template.
 | **dapi.changeDefaults.server_spread**        | String | -            | **DEPRECATED** How VMs are spread across CNs (one of: min-ram, max-ram, min-owner, and random)   |
 | **dapi.changeDefaults.filter_docker_min_platform** | String | -      | If present, minimum platform version useful for Docker instances.        |
 | **dapi.changeDefaults.filter_flexible_disk_min_platform** | String | - | If present, minimum platform version useful for instances with flexible disk sizing. |
+| **dapi.changeDefaults.filter_capness**      | String | true         | Whether to disallow mixing of instances that set cpu_caps with instances that do not set cpu_caps on the same compute node. |
 | **dapi.changeDefaults.filter_headnode**      | String | true         | Whether VMs cannot allocate on the headnode.                             |
 | **dapi.changeDefaults.filter_min_resources** | String | true         | Whether CPU/RAM/disk limits are ignored when allocating.                 |
 | **dapi.changeDefaults.filter_large_servers** | String | true         | Whether large servers are reserved for larger allocations.               |
@@ -107,6 +108,7 @@ specialized circumstances in production.
 | Key                            | Type    | Default | Description                                                                  |
 | ------------------------------ | ------- | ------- | ---------------------------------------------------------------------------- |
 | **ALLOC_SERVER_SPREAD**        | String  | -       | **DEPRECATED** How the allocator spreads VMs across CNs.                     |
+| **ALLOC_FILTER_CAPNESS**       | Boolean | true    | Whether to disallow mixing of instances that set cpu_caps with instances that do not set cpu_caps on the same compute node. |
 | **ALLOC_FILTER_HEADNODE**      | Boolean | true    | Whether the headnode should be removed from consideration during allocation. |
 | **ALLOC_FILTER_MIN_DISK**      | Boolean | false   | Whether CNs with insufficient spare disk should be removed.                  |
 | **ALLOC_FILTER_MIN_RESOURCES** | Boolean | true    | Whether CNs with insufficient spare CPU/RAM/disk should be removed.          |
diff --git a/docs/static.md b/docs/static.md
index 8c8895b..3c70711 100644
--- a/docs/static.md
+++ b/docs/static.md
@@ -42,6 +42,7 @@ found in sapi_manifests/cnapi/template.
 | **dapi.changeDefaults.server_spread**        | String | -            | **DEPRECATED** How VMs are spread across CNs (one of: min-ram, max-ram, min-owner, and random)   |
 | **dapi.changeDefaults.filter_docker_min_platform** | String | -      | If present, minimum platform version useful for Docker instances.        |
 | **dapi.changeDefaults.filter_flexible_disk_min_platform** | String | - | If present, minimum platform version useful for instances with flexible disk sizing. |
+| **dapi.changeDefaults.filter_capness**      | String | true         | Whether to disallow mixing of instances that set cpu_caps with instances that do not set cpu_caps on the same compute node. |
 | **dapi.changeDefaults.filter_headnode**      | String | true         | Whether VMs cannot allocate on the headnode.                             |
 | **dapi.changeDefaults.filter_min_resources** | String | true         | Whether CPU/RAM/disk limits are ignored when allocating.                 |
 | **dapi.changeDefaults.filter_large_servers** | String | true         | Whether large servers are reserved for larger allocations.               |
@@ -76,6 +77,7 @@ specialized circumstances in production.
 | Key                            | Type    | Default | Description                                                                  |
 | ------------------------------ | ------- | ------- | ---------------------------------------------------------------------------- |
 | **ALLOC_SERVER_SPREAD**        | String  | -       | **DEPRECATED** How the allocator spreads VMs across CNs.                     |
+| **ALLOC_FILTER_CAPNESS**       | Boolean | true    | Whether to disallow mixing of instances that set cpu_caps with instances that do not set cpu_caps on the same compute node. |
 | **ALLOC_FILTER_HEADNODE**      | Boolean | true    | Whether the headnode should be removed from consideration during allocation. |
 | **ALLOC_FILTER_MIN_DISK**      | Boolean | false   | Whether CNs with insufficient spare disk should be removed.                  |
 | **ALLOC_FILTER_MIN_RESOURCES** | Boolean | true    | Whether CNs with insufficient spare CPU/RAM/disk should be removed.          |
diff --git a/lib/designation.js b/lib/designation.js
index e8d5a56..9f70b2a 100644
--- a/lib/designation.js
+++ b/lib/designation.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2019, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -26,6 +26,7 @@ var DEFAULT_WEIGHT_UNIFORM_RANDOM   = 0.5;
 var DEFAULT_WEIGHT_UNRESERVED_DISK  = 1;
 var DEFAULT_WEIGHT_UNRESERVED_RAM   = 2;
 
+var DEFAULT_FILTER_CAPNESS       = true;
 var DEFAULT_FILTER_HEADNODE      = true;
 var DEFAULT_FILTER_MIN_RESOURCES = true;
 var DEFAULT_FILTER_LARGE_SERVERS = true;
@@ -153,6 +154,7 @@ function getDefaults(changeDefaults) {
 
     setDefault('disable_override_overprovisioning',
                DEFAULT_DISABLE_OVERRIDE_OVERPROV);
+    setDefault('filter_capness',      DEFAULT_FILTER_CAPNESS);
     setDefault('filter_headnode',      DEFAULT_FILTER_HEADNODE);
     setDefault('filter_min_resources', DEFAULT_FILTER_MIN_RESOURCES);
     setDefault('filter_large_servers', DEFAULT_FILTER_LARGE_SERVERS);
diff --git a/package.json b/package.json
index 60037f2..ec82705 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "Triton Compute Node API",
-  "version": "1.24.2",
+  "version": "1.24.3",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
@@ -10,7 +10,7 @@
     "async": "1.5.2",
     "backoff": "2.5.0",
     "bunyan": "1.8.5",
-    "dapi": "git+https://github.com/joyent/sdc-designation.git#858790422cc58d439ec29c5ea403dc471e56fa89",
+    "dapi": "git+https://github.com/jlevon/sdc-designation.git#f9f5b7fec54474fedf67df20bd1ae1e052f876c2",
     "deep-object-diff": "1.1.0",
     "dox": "0.9.0",
     "gc-stats": "1.2.0",
diff --git a/sapi_manifests/cnapi/template b/sapi_manifests/cnapi/template
index 222823f..0dc3b89 100644
--- a/sapi_manifests/cnapi/template
+++ b/sapi_manifests/cnapi/template
@@ -52,6 +52,7 @@
 	"dapi": {
 		"changeDefaults": {
 			"server_spread": "{{{ALLOC_SERVER_SPREAD}}}",
+			"filter_capness": "{{{ALLOC_FILTER_CAPNESS}}}",
 			"filter_headnode": "{{{ALLOC_FILTER_HEADNODE}}}",
 			"filter_min_resources": "{{{ALLOC_FILTER_MIN_RESOURCES}}}",
 			"filter_large_servers": "{{{ALLOC_FILTER_LARGE_SERVERS}}}",
